src/test/scala/firrtlTests/CustomTransformSpec.scala

```diff
@@ -71,5 +73,28 @@ class CustomTransformSpec extends FirrtlFlatSpec {
       Driver.execute(optionsManager)
     }).getMessage should include (errorString)
   }
-}
 
+  object Foo {
+    class A extends Transform {
+      def inputForm = HighForm
+      def outputForm = HighForm
+      def execute(s: CircuitState) = {
+        println(name)
```

Comment:
Seems like this would be a nicer test and make clear to code reader what the new name format looks like
 if you got replaced the `println()` with
`assert(name === "firrtlTests.CustomTransformSpec$Foo$A")`

Reply:
I did appreciate your XCF structure in PR name

Reply:
I'm reluctant to do that as such a test is really a check on the exact functionality of Java's `getName` method in a Scala environment. That should be allowed to change independent of the FIRRTL code base. The alternative of doing `assert(name == this.getClass.getName)` would seem to be useless...

What do you think?

This test is stripped down from what I originally had (and was exactly what was causing the problem). I had a custom transform inside an object that deleted annotations. The `propagateAnnotations` method calls the `name` method and will except similarly (though more opaquely). This would be a more opaque way of testing the same thing:

```scala
  object Foo {
    case object A extends NoTargetAnnotation
    class B extends Transform {
      def inputForm = HighForm
      def outputForm = HighForm
      private def strip(a: Annotation): Boolean = a match {
        case A => false
        case _ => true
      }
      def execute(s: CircuitState) = s.copy(annotations = s.annotations.filter(strip))
    }
  }

  they should "work if placed inside an object" in {
    val input =
      """|circuit Foo:
         |  module Foo:
         |    node a = UInt<1>(0)
         |""".stripMargin
    val annotations = Seq(
      Foo.A,
      RunFirrtlTransformAnnotation(new Foo.B),
      FirrtlSourceAnnotation(input) )

    (new FirrtlStage).execute(Array.empty, annotatinos)
  }
```

Reply:
Presumably we would catch such a change some other way, right? Any java changes that we can catch we should because it could affect users

Reply:
I can see benefits to both. Generally, I agree with the "conservative test failing as informative signal rather than true bug" philosophy, but I don't know how well that extends to things that only get printed in debug output.

I agree with Schuyler that even though Scala class name mangling and nesting is stable within a major version, it changes a lot across major versions, especially for anonymous values.